src/io/io_behaviorsaver.js

```diff
@@ -58,13 +58,89 @@ IO.BehaviorSaver = new (function() {
 		}
 	}
 
+    String.prototype.format = function() {
+        var formatted = this;
+        for( var arg in arguments ) {
+            formatted = formatted.replace("{" + arg + "}", arguments[arg]);
+        }
+        return formatted;
+    };
+
 	var saveSuccessCallback = function() {
 		T.logInfo("Save successful!");
 		UI.Panels.Terminal.hide();
 		UI.Settings.updateBehaviorlib();
 		UI.Tools.notifyRosCommand('save');
-	}
+    }
+
+    var saveOnRobot = function() {
+        var names = Behavior.createNames();
+        var package_name = names.rosnode_name;
+        var ssh_username = UI.Settings.getRobotUsername();
+        var ssh_hostname = UI.Settings.getRobotHostname();
+        var ssh_package_path = ""
+
+        if(ssh_username === "" || ssh_hostname === "" /*|| ssh_password === ""*/) {
+            T.logError("SSH configurations are not well defined, fix them from Configuration Tab -> Code Generation Box");
+            return;
+        }
+
+        var ssh_password = window.prompt("Please enter {0}:{1} password: ".format(ssh_username, ssh_hostname));
+        if(ssh_password == null || ssh_password == "") {
+            return;
+        }
 
+        ROS.getPackagePath(package_name, (package_path) =>
+        {
+            try
+            {
+                var SSHClient = require('ssh2').Client;
```

Comment:
Shouldn't ssh2 be an npm dependency as well? Even if included in scp2, it is referenced here directly.